### ISSUE TITLE
Unexpected error window that says HTTP 304 code came up

### ISSUE THREAD

**Issue by KENNYSOFT** [desktop/desktop#3445]
### Description

Unexpected error window that says HTTP 304 (Not modified) code came up. I think HTTP 304 (Not modified) should be treated internally.

### Version

**GitHub Desktop version:** 1.0.9

**OS version:** Microsoft Windows [Version 10.0.16299.64]

### Steps to Reproduce

1. _I cannot reproduce_

**Expected behavior:** HTTP 304 treated internally and go as normally

**Actual behavior:** Error window came up

**Reproduces how often:** Only once

### Logs

```
2017-11-28T08:02:08.153Z - info: [ui] [AppStore] loading 11 repositories from store
2017-11-28T08:02:08.161Z - info: [ui] [AppStore] found account: KENNYSOFT (KENNYSOFT)
2017-11-28T08:02:09.899Z - warn: [ui] Unable to deserialize JSON string to object [object Response]
SyntaxError: Unexpected end of JSON input
2017-11-28T08:02:10.338Z - warn: [ui] fetchPullRequests: failed for repository kennysoft/MIPS_design
Error: API error https://api.github.com/repos/kennysoft/MIPS_design/pulls?state=open&per_page=100: Not Modified (304)
    at Object.t.parsedResponse (C:\Users\525hm\webpack:\app\src\lib\http.ts:148:34)
    at <anonymous>
2017-11-28T08:02:10.339Z - warn: [ui] Error refreshing pull requests for 'MIPS_design'
Error: API error https://api.github.com/repos/kennysoft/MIPS_design/pulls?state=open&per_page=100: Not Modified (304)
    at Object.t.parsedResponse (C:\Users\525hm\webpack:\app\src\lib\http.ts:148:34)
    at <anonymous>
2017-11-28T08:02:10.549Z - info: [ui] launching: 1.0.9 (Windows 10.0.16299)
```

#### Additional Information

![image](https://user-images.githubusercontent.com/6624567/33319370-bd93bc94-d481-11e7-8aef-61510d33bb12.png)

**Comment by j-f1** [comment]
We’re working on getting this fixed in #3421, and the fix for it should hopefully be released in the next update.

**Comment by joshaber** [comment]
#3421 has landed and will be out in the next release.
